### ISSUE TITLE
Cannot be serialized

### ISSUE THREAD

**Issue by chekanov** [fasseg/exp4j#48]
Hello, 

The current of exp4j breaks the serialization mechanism of the DMelt project (http:/jwork.org/dmelt/).
For example, the objects created by the class Expression and ExpressionBuilder cannot be serialized
(java.io.NotSerializableException) due to the keys used to create maps inside exp4j. This problem exists after adding " implements java.io.Serializable" statements for Expression and ExpressionBuilder classes.

best, Sergei

**Comment by leogtzr** [comment]
Have you tried adding "implements Serializable" to the remaining classes?
I agree that we should implement from Serializable.

**Comment by chekanov** [comment]
Yes, of course.  I've  added "implements Serializable" to Expression and ExpressionBuilder.  It still cannot be serialized. It has a long sting of exceptions that seems related to the unserializable keys of the map used to evaluate expression.

**Comment by leogtzr** [comment]
You have to do the same with Function, Operator, etc ...

**Comment by fasseg** [comment]
@chekanov: Currently this is out of scope for exp4j. But I'll leave this feature request open if someone wants to chime in and supply a PR.
